src/main/java/teammates/storage/api/CommentsDb.java

```diff
@@ -199,92 +126,40 @@ public CommentAttributes getComment(CommentAttributes commentToGet) {
         Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, commentViewerType);
         
         List<Comment> comments = getCommentEntitiesForCommentViewer(courseId, commentViewerType);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return getAttributesListFromEntitiesList(comments);
     }
     
-    /*
-     * Get comment for the sending state (SENT|SENDING|PENDING)
-     */
+    @SuppressWarnings("unchecked")
     public List<CommentAttributes> getCommentsForSendingState(String courseId, CommentSendingState state) {
-        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
-        
-        List<Comment> comments = getCommentEntitiesForSendingState(courseId, state);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return (List<CommentAttributes>) super.getCommentsForSendingState(courseId, "PLACEHOLDER", state);
     }
     
     /*
      * Get comments for a course
      */
+    @SuppressWarnings("unchecked")
     public List<CommentAttributes> getCommentsForCourse(String courseId) {
-        Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, courseId);
-        
-        List<Comment> comments = getCommentEntitiesForCourse(courseId);
-        List<CommentAttributes> commentAttributesList = new ArrayList<CommentAttributes>();
-        
-        for (Comment comment : comments) {
-            commentAttributesList.add(new CommentAttributes(comment));
-        }
-        return commentAttributesList;
+        return (List<CommentAttributes>) super.getCommentsForCourse(courseId);
```

Comment:
I tried making an abstract conversion method and placed them in Base(FR)CommentsDb/Logic before each return, but it didn't help because for each of this kind of method, the super method's return type is still the superclass, or list of superclass.
Conversion will work if it's placed in the subclass, but it's pretty much what's already been done.